### PR TITLE
refactor: streamlines LabelList and KCard usage

### PR DESCRIPTION
**Pull request by kleinfreund**
Removes the built-in KCard from LabelList so that it can be (1) nested and (2) used where no card styles are necessary (e.g. in a context where this already a wrapping KCard).

Integrates a default KCard into the `TabsWidget` tabs and removes all instances where one was explicitly added. This cuts down on quite a bit of redundant code.

Uses KCard in some place where we were previously achieving a similar look with class-based padding. Overall, these changes achieve a much more consistent padding of panel-like boxes in the UI.

Removes some unused actions columns from some tables.

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

### CODE REVIEW

File: src/assets/styles/_kongponents-overrides.scss
Comment (kleinfreund) [reply src/assets/styles/_kongponents-overrides.scss:17]: I’m using `inherit` here so that this element inherits the font size and line height from parent elements which has the effect that it uses the default font sizes _we use in the project_ (i.e. not browser defaults). The same applies to `currentColor` and in this case `inherit`.

With the default styles of the card applied, the font size of card content is set to 13 pixels by default and in most cases, that looks too small to me and breaks out of the general styles of the project. Generally, it shouldn’t be a component library being opinionated about how the base styles of a project look. This can be part of a design system or a project’s custom definitions, but should be kept out of components who should just adapt to such defaults.

The difference is easier observed by going to a page with a KCard component, selecting the element with the `k-card-body` class, and toggling the three override property declarations in the Styles panel.

---

Example: https://deploy-preview-754--kuma-gui.netlify.app/gui/zoneegresses?zoneEgress=zoneegress-1&tab=insights

**Currently**:

Screenshot from 2023-03-24 09-50-47

**In this PR**:

Screenshot from 2023-03-24 09-50-55

---

Example: https://deploy-preview-754--kuma-gui.netlify.app/gui/

**Currently**:

Screenshot from 2023-03-24 09-57-43

**In this PR**:

Screenshot from 2023-03-24 09-57-35

Comment (johncowen) [reply src/assets/styles/_kongponents-overrides.scss:17]: ty! I'm just reading over/looking at this in more detail.

Just while I'm doing that I wanted to ask about the first set of screengrabs where the second level headers are now bigger than the first level headers, here:

Screenshot 2023-03-24 at 10 24 25

Are we ok with this?

Comment (kleinfreund) [reply src/assets/styles/_kongponents-overrides.scss:17]: True, that is not good.

The heading hierarchy is broken in this view for starters. I use the following naive script to get a document’s heading hierarchy:

```js
console.log(Array.from(document.querySelectorAll('h1,h2,h3,h4,h5,h6')).map((h) => ('-'.repeat(parseInt(h.tagName[1]))) + ` ${h.textContent.trim()}`).join('\n'))
```

Note how we skip levels 2 and 3 entirely here.

```
- Zone Egress: zoneegress-1
---- Connect time: July 13, 2021 at 10:41:04 AM
----- General Information:
------ cds:
------ eds:
------ lds:
------ rds:
```

If I fix this and remove the font size override from the too small headings here (e.g. General Information:):

```
- Zone Egress: zoneegress-1
-- Connect time: July 13, 2021 at 10:41:04 AM
--- General Information:
---- cds:
---- eds:
---- lds:
```

image

(I also fixed the issue of sometimes showing a heading when we don’t have a value to display).

**Comment by johncowen** [comment]
Looking at these two side by side (this PR on the left, previous to this PR on the right), the differences are really striking, probably open that image up fullsize to get a real impression)

Screenshot 2023-03-24 at 16 32 21

I know you are waiting for this work to continue on other things, so I'm half in mind to approve this despite the fact that I'm really not sure about all these changes in font sizes etc. Moreover, I've been noticing little changes in font sizes etc with previous PRs, and whilst the GUI isn't super duper polished in that perspective I do wonder what effect these extra visual changes are having.

It's also a bit difficult as I don't fully follow what was required in this PR for the changes you are working on. e.g. if I knew this wasn't potentially holding up other work I would probably query these visual changes more.

Just trying to think of how we can avoid the quibbles over the visuals. I think code-wise and functionality-wise I'd be happy to approve this now, just the visual changes are what's making me wonder. Just incase it makes this easier, is there anyway to hold off on the visual changes in this PR or, maybe just check in with say @lahabana to get a second opinion on the visuals., I wouldn't want to prevent approval on those if others where like "this looks fine to me".

**Comment by kleinfreund** [comment]
First of all, I definitely don't want to make this a blocker. This work doesn't have a lot of priority so we can take some time here.

I do realize that this isn't strictly speaking about the usage of LabelList and KCard, but has changes to related areas of the UI that can broadly be categorized as typography. I think the current state is really quite bad with heading structures being out of order and a quite a few varying font sizes being applied for no apparent reason. So I definitely want to work on this whenever I find the time because this is the kind of stuff that never gets looked via any other way.

**I'll try and extract typographical changes into a separate PR so that this PR is mostly scoped to what the title says, but I do want to continue this work because it otherwise doesn't happen.**

**Comment by johncowen** [comment]
Gotcha, ok great that sounds super good if you could do that, I think that'd help 👍

**Comment by kleinfreund** [comment]
@lahabana re: https://github.com/kumahq/kuma-gui/pull/754#issuecomment-1484879371

That makes sense and I agree with what you stated. To turn this into a few more practical items for us to do:

- **Good baseline typography**: we should define our base body copy font size and the font sizes of the 6 heading levels and then override them in components as little as possible. This also implies that Kongponents-inherent overrides (e.g. like `KCard`’s) need to be undone. (While we do this, we should also fix the broken heading structures on some of our places.)
- **Consistent grouping of content in panels/cards**: We should consistently place content in panels (i.e. `KCard`s). in some places, content appears outside of panels (most notably: the home page and the meshes overview).
- **Design generic and flexible definition list component**: We need to take `LabelList` and turn it into a more flexible and generic `DefinitionList` component and use that wherever we display the fields of resources. We currently don’t use `LabelList` everywhere and it doesn’t accommodate all use cases equally well.
- (**Separate lists from summaries/details**: In views where we display both a list of resources and a summary of a selected row, we should display the list and the summary in a separate panel to clearly indicate this boundary visually.)

**Review by johncowen — APPROVED** [review]
Whilst there are some minor visual changes here, we had a chat about about it so this is good to go!! 🚂

**Pull request merged**
